src/e2e-test/java/io/cdap/plugin/odp/stepsdesign/RuntimeODP.java

```diff
@@ -0,0 +1,422 @@
+                                        String splitRow, String packageSize) throws IOException, InterruptedException {
+    datasource = CDAPUtils.getPluginProp(dsName);
+    ODPActions.getODPProperties();
+    ODPActions.enterDirectConnectionProperties(client, sysnr, asHost, dsName, gcsPath, splitRow, packageSize);
+  }
+
+  @When(
+    "Configure Connection {string} {string} {string} {string} {string} {string} {string} {string} {string} {string}")
+  public void configureConnection(String client, String sysnr, String asHost, String dsName, String gcsPath,
+                                  String splitRow, String packageSize, String msServ, String systemID,
+                                  String lgrp) throws IOException, InterruptedException {
+    ODPActions.getODPProperties();
+    ODPActions.enterConnectionProperties(client, sysnr, asHost, dsName, gcsPath, splitRow, packageSize,
+                                         msServ, systemID, lgrp);
+  }
+
+  @When("Username and Password is provided")
+  public void usernameAndPasswordIsProvided() throws IOException {
+    ODPActions.enterUserNamePassword(CDAPUtils.getPluginProp("AUTH_USERNAME"),
+                                     CDAPUtils.getPluginProp("AUTH_PASSWORD"));
+  }
+
+  @When("Run one Mode is Sync mode")
+  public void runOneModeIsSyncMode() {
+    ODPActions.selectSync();
+  }
+
+  @Then("Validate the Schema created")
+  public void validateTheSchemaCreated() throws InterruptedException {
+    ODPActions.getSchema();
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 20000);
+    wait.until(ExpectedConditions.numberOfElementsToBeMoreThan
+      (By.xpath("//*[@placeholder=\"Field name\"]"), 2));
+    Assert.assertEquals(true, CDAPUtils.schemaValidation());
+  }
+
+  @Then("Close the ODP Properties")
+  public void closeTheODPProperties() {
+    ODPActions.closeButton();
+  }
+
+  @Then("Delete the existing Odp table {string} in bigquery")
+  public void deleteTheExistingOdpTableInBigquery(String table) throws IOException, InterruptedException {
+    gcpClient.dropBqQuery(CDAPUtils.getPluginProp(table));
+    BeforeActions.scenario.write("Table Deleted Successfully");
+  }
+
+  @Then("link source and target")
+  public void linkSourceAndTarget() {
+    SeleniumHelper.dragAndDrop(ODPLocators.fromODP, CdfStudioLocators.toBigQiery);
+  }
+
+  @Then("Delete all existing records in datasource")
+  public void deleteAllExistingRecordsInDatasource() throws JCoException, IOException {
+    sapProps = SAPProperties.getDefault(connection);
+    errorCapture = new ErrorCapture(exceptionUtils);
+    sapAdapterImpl = new SAPAdapterImpl(errorCapture, connection);
+    Map opProps = new HashMap<>();
+    opProps.put("RFC", "ZTEST_DATA_DEL_ACDOCA");
+    opProps.put("autoCommit", "true");
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      ObjectNode objectNode = mapper.createObjectNode();
+      objectNode.put("I_ALL", "X");
```

Comment:
what is this? please add comments explaining what this means

Reply:
code unused, so removed